app/services/update_account_service.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class UpdateAccountService
+  def initialize(account, params)
+    @account = account
+    @params = params
+    @account_share = @account.account_shares.where.not(parental_key: nil).first
+
+    return unless @account_share
+
+    @kid_user = User.includes(:account).find_by(parental_key: @account_share.parental_key)
+    @kid_account = @kid_user&.account
+  end
+
+  def perform
+    return false if email_already_exists?(@params[:email])
```

Comment:
Should it be a validation with a clear error? Also, WDYT about this name - `another_kid_user_exists?(email)` - is it descriptive enough to explain what it's doing?